Extension/src/LanguageServer/configurations.ts

```diff
@@ -243,6 +243,33 @@ export class CppProperties {
         vscode.workspace.onDidChangeTextDocument((e: vscode.TextDocumentChangeEvent) => {
```

Comment:
Hmm, this existing code seems incorrect. It's being called on every document edit instead of on save. I think there's an existing bug where c_cpp_properties.json edits are being handled before the file is saved.

Reply:
I think this should be changed to "vscode.workspace.onDidSaveTextDocument((doc: vscode.TextDocument) => {``

Reply:
Actually, this code may have been intended to handle file creation/deletion as well...but that's already handled by the file watchers.

It might be better to handle this in the a didSave handler called on the client (i.e. add one to protocolFilter.txt).

Reply:
Yes my mistake. I didn't realize I was in onDidChange somehow I thought I was in onDidSave all the time like the counterpart fix in CMake Tools. I am correcting this in an update.